docs/documentation/native/native2.mdx

```diff
@@ -0,0 +1,115 @@
+- **Flaky text selectors**: Text-based selectors can be unreliable as they may change with localization or UI updates
+- **Platform incompatibility**: A single selector approach couldn't effectively handle the fundamental differences between iOS and Android element identification
+
+Native2 provides a **single command** that can work with both Android and iOS selectors, eliminating the need to write separate platform-specific code. Instead of maintaining different methods for each platform, you can now use one method that accepts platform-specific selectors that acknowledge the inherent differences between platforms.
+
+
+### Before native2 (Separate Methods)
+
+```dart
+// You were forced to use flaky text-based selectors that work on both platforms
+await $.native.tap(Selector(textContains: 'Login'));
+```
+
+```dart
+// Before: Sometimes you needed to use platform-specific if statements in your test code.
+if (Platform.isAndroid) {
+  await $.native.tap(Selector(resourceId: 'com.android.camera2:id/shutter_button'));
+} else {
+  await $.native.tap(Selector(label: 'Take Picture'));
+}
+```
+
+### With native2
+
+Native2 provides a single method that works across both platforms:
+
+```dart
+// After: Single method with platform-specific selectors
+await $.native2.tap(
+  NativeSelector(
+    android: AndroidSelector(
+      resourceName: 'com.android.camera2:id/shutter_button',
+    ),
+    ios: IOSSelector(label: 'Take Picture'),
+  ),
+);
+
+// Text input with unified method
+await $.native2.enterText(
+  NativeSelector(
+    android: AndroidSelector(
+      contentDescription: 'Email field',
+    ),
+    ios: IOSSelector(
+      identifier: 'emailField',
+    ),
+  ),
+  text: 'john@example.com',
+);
+```
+
+### Text Input Operations
+
+```dart
+// Enter password in secure field
+await $.native2.enterText(
+  NativeSelector(
+    android: AndroidSelector(
+      contentDescription: 'Password',
+    ),
+    ios: IOSSelector(
+      elementType: IOSElementType.secureTextField,
+    ),
+  ),
+  text: 'secretpassword',
+);
+```
+
+### More platform-specific attributes like elementType for iOS
+
+```dart
+// Find elements by instance (when multiple elements match)
+await $.native2.tap(
+  NativeSelector(
+    android: AndroidSelector(
+      className: 'android.widget.Button',
+      instance: 2, // Third button (0-indexed)
+    ),
+    ios: IOSSelector(
+      elementType: IOSElementType.button,
+      instance: 2, // Third button (0-indexed)
+    ),
+  ),
+);
+```
+
+### Specifying App ID (iOS)
+
+When working with iOS, you may need to specify the `appId` parameter to interact with elements in specific applications. This is particularly useful when your test needs to interact with system apps like Safari, Settings, or other third-party applications.
+
+<Info>
+The `appId` parameter is primarily needed for iOS. On Android, the app context is usually handled automatically.
```

Comment:
I think that appId parameter is ignored on Android - if that's true, it's worth writing about it here